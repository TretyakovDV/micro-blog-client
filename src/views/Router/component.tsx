import React from 'react';
import { inject, observer } from 'mobx-react';
import { Switch, Route, Redirect } from 'react-router-dom';

import Home from 'views/Home';
import Post from 'views/Post';
import AddPost from 'views/AddPost';
import Login from 'views/Login';
import Header from 'views/Header';
import { User } from 'stores/user';

type Props = {
  user?: User
};

const Component = inject('user')(observer(({ user }: Props) => (
  <>
    <Header />
    {user?.auth ? (
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route exact path="/post/:id">
          <Post />
        </Route>
        <Route exact path="/login">
          <Login />
        </Route>
        <Route exact path="/add-post">
          <AddPost />
        </Route>
        <Redirect exact from="*" to="/" />
      </Switch>
    ) : (
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route exact path="/post/:id">
          <Post />
        </Route>
        <Route exact path="/login">
          <Login />
        </Route>
        <Redirect exact from="*" to="/" />
      </Switch>
    )}

  </>
)));

export default Component;
