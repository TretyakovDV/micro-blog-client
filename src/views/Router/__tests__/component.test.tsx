import React from 'react';
import { render } from '@testing-library/react';

import Router from '../component';

describe('Router', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Router />);
    expect(asFragment()).toMatchSnapshot();
  });
});
