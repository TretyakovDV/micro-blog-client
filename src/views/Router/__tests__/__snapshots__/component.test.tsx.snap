// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Component
  </p>
</DocumentFragment>
`;
